sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt

```diff
@@ -741,16 +743,23 @@ class CompactBlockProcessor internal constructor(
     /**
      * Refreshes the SDK's wallet summary from the Rust backend, and transmits this information
      * into the related internal flows. Note that the Orchard balance is not yet supported.
+     *
+     * @return the next subtree index to fetch.
      */
-    internal suspend fun refreshWalletSummary() {
+    internal suspend fun refreshWalletSummary(): UInt {
         when (val result = getWalletSummary(backend)) {
             is GetWalletSummaryResult.Success -> {
                 val resultProgress = result.scanProgressPercentDecimal()
                 Twig.info { "Progress from rust: ${resultProgress.decimal}" }
                 setProgress(resultProgress)
                 updateAllBalances(result.walletSummary)
+                return result.walletSummary.nextSaplingSubtreeIndex
+            }
+            else -> {
+                // Do not report the progress and balances in case of any error, and
+                // tell the caller to fetch all subtree roots.
+                return UInt.MIN_VALUE
```

Comment:
@HonzaR This seems like a placeholder; we should really have more precise error handling here, otherwise it seems like this could end up "stuck at zero"; we do want to handle an error here if the wallet is completely uninitialized, but any other error should be handled more carefully.

Reply:
Created #1374